packages/messageport/lib/messageport_tizen.dart

```diff
@@ -0,0 +1,163 @@
+  /// Exception will be thrown when using on already registered port
+  void register(OnMessageReceived onMessage) {
+    if (_registeredPorts.contains(portName)) {
+      throw Exception('Port $portName is already registered');
+    }
+
+    _eventChannel = EventChannel('tizen/messageport$_id');
+    _streamSubscription = _eventChannel.receiveBroadcastStream().listen((map) {
+      if (map is Map) {
+        Object message = map['message'];
+        if (map.containsKey('remotePort')) {
+          String remoteAppId = map['remoteAppId'];
+          String remotePort = map['remotePort'];
+          bool isTrusted = map["isTrusted"];
+          onMessage(
+              message, TizenRemotePort._(remoteAppId, remotePort, isTrusted));
+        } else {
+          onMessage(message);
+        }
+      }
+    });
+
+    _registeredPorts.add(portName);
+  }
+
+  /// Unregisters messageport
+  Future<void> unregister() async {
+    await _streamSubscription?.cancel();
+    _registeredPorts.remove(portName);
+  }
+
+  /// Checks whether local port is trusted
+  bool get trusted {
+    return _isTrusted;
+  }
+
+  // Returns port name
+  String get portName {
+    return _portName;
+  }
+
+  /// Checks whether local port is registered
+  bool get registered {
+    return _registeredPorts.contains(_portName);
+  }
+
+  final int _id;
+  final bool _isTrusted;
+  bool _isRegistered = false;
+  String _portName;
+  StreamSubscription<Object> _streamSubscription;
+  EventChannel _eventChannel;
+
+  static Set<String> _registeredPorts = Set<String>();
+}
+
+/// Remote port to send messages
+class TizenRemotePort {
+  TizenRemotePort._(this._remoteAppId, this._portName, this._isTrusted);
+
+  /// Sends message through remote messageport
+  Future<void> send(Object message) async {
+    Map<String, Object> args = new Map<String, Object>();
+    args["remoteAppId"] = _remoteAppId;
+    args["portName"] = _portName;
+    args["message"] = message;
+
+    return _channel.invokeMethod("send", args);
+  }
+
+  /// Sends message through remote messageport with local port
+  ///
+  /// Remote application can reply to the message by use of provided local port
+  Future<void> sendWithLocalPort(
+      Object message, TizenLocalPort localPort) async {
+    Map<String, Object> args = new Map<String, Object>();
+    args["remoteAppId"] = _remoteAppId;
+    args["portName"] = _portName;
+    args["localPort"] = localPort._id;
+    args["message"] = message;
+
+    return _channel.invokeMethod("send", args);
+  }
+
+  /// Returns remote appId
+  String get remoteAppId {
+    return _remoteAppId;
+  }
+
+  /// Returns port name
+  String get portName {
+    return _portName;
+  }
+
+  /// Checks whether remote port is trusted
+  bool get trusted {
+    return _isTrusted;
+  }
+
+  final String _remoteAppId;
+  final String _portName;
+  final bool _isTrusted;
+}
+
+/// API for accessing MessagePorts in Tizen
+class TizenMessageport {
```

Comment:
How do you think of removing the `Tizen` prefix from all class names? I think it's redundant.

Reply:
I can do it but `MessagePort` is already a part of [dart:html](https://api.dart.dev/stable/2.13.1/dart-html/MessagePort-class.html). If it's OK to have `Messageport` (from this plugin) and `MessagePort` (from html), then I can change the name. Or should I change it `MessagePort`?

Reply:
Oops. In that case, it'd be better to use the name `TizenMessagePort` (capitalized P) to avoid confusion, but you may still consider removing `Tizen` from `TizenLocalPort` and `TizenRemotePort`.